### PR TITLE
Convert the release script to python

### LINKED ISSUE #43: Rewrite release scripts in Python
This would make it easier to test and IMO the bash scripts are getting a little big to remain readable.

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #43 

#### What's this PR do?
Rewrites the release scripts in Python and adds some tests

#### How should this be manually tested?
Run a release. The related shell scripts will forward to the new python scripts and they should have full backwards compatibility

### CODE REVIEW

**Comment by noisecapella** [comment]
Sounds good. @mbertrand can you use this branch when you do the next MM release?

**Comment by mbertrand** [comment]
Sure, I'll switch over now

File: release.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+    raise OldVersionException("Unable to find previous version number")
+
+
+def create_release_notes(old_version):
+    """Returns the release note text for the commits made for this version"""
+    return check_output([
+        "git-release-notes",
+        "v{}..master".format(old_version),
+        os.path.join(SCRIPT_DIR, "util/release_notes_rst.ejs"),
```
Comment (rhysyngsun) [inline release.py]: Needs to use ``release_notes.ejs` for the PR body text

File: finish_release.sh [hunk 2]
```diff
@@ -1,54 +1,11 @@
...
-    create_working_dir
-    check_release_tag
-    merge_release_candidate
-    tag_release
-    merge_release
-    delete_working_dir
-}
-
-if [[ $(basename $0) = "finish_release.sh" ]]; then
+if [[ $(basename $0) = "release.sh" ]]; then
```
Comment (rhysyngsun) [inline finish_release.sh]: This should be `finish_release.sh`

File: finish_release.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    check_call(['git', 'checkout', '-t', 'origin/release'])
+    check_call(['git', 'merge', 'origin/release-candidate'])
+    check_call(['git', 'push'])
+
+
+def check_release_tag(version):
+    """Check release version number"""
+    print("Check release version number...")
+    check_call(['git', 'checkout', '-t', 'origin/release-candidate'])
+    commit_name = check_output(['git', 'log', '-1', '--pretty=%B']).decode()
```
Comment (rhysyngsun) [inline finish_release.py]: Add `.strip()` since the output can have newlines/whitespace in it.

File: release.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+    release_filename = "RELEASE.rst"
+    try:
+        with open(release_filename) as f:
+            existing_note_lines = [line for line in f.readlines()]
+    except FileNotFoundError:
+        existing_note_lines = []
+
+    with open(release_filename, "w") as f:
+        f.write("Release notes\n")
```
Comment (rhysyngsun) [inline release.py]: Could you capitalize "notes" so it matches the previous template?

Comment (noisecapella) [reply release.py]: Yes

**Pull request merged**